files/src/main/java/com/fincity/saas/files/service/AbstractFilesResourceService.java

```diff
@@ -739,6 +745,290 @@ public Mono<FileDetail> create(String clientCode, String uri, FilePart fp, Strin
 				.contextWrite(Context.of(LogUtil.METHOD_NAME, "AbstractFilesResourceService.create"));
 	}
 
+	public Mono<FileDetail> imageUpload(String clientCode, String uri, FilePart fp, String fileName, Boolean override,
+			ImageDetails imageDetails, String filePath) {
+		boolean ovr = override == null || override.booleanValue();
+		
+		// just removing the TRANSFORM_TYPE from the uri
+		if (uri.indexOf(TRANSFORM_TYPE) != -1) {
+			int ind = uri.indexOf(TRANSFORM_TYPE);
+			uri = uri.substring(0, ind) + uri.substring(ind + TRANSFORM_TYPE.length() + 1);
+		}
+		
+		Tuple2<String, String> tup = this.resolvePathWithoutClientCode(this.uriPart, uri);
+		String resourcePath = tup.getT1();
+		String urlResourcePath = tup.getT2();
+		
+		return FlatMapUtil.flatMapMonoWithNull(
+
+				() -> this.fileAccessService.hasWriteAccess(resourcePath, clientCode, this.getResourceType()),
+
+				hasPermission -> {
+					if (!hasPermission.booleanValue())
+						return msgService.throwMessage(msg -> new GenericException(HttpStatus.FORBIDDEN, msg),
+								FilesMessageResourceService.FORBIDDEN_PATH, this.getResourceType(), resourcePath);
+
+					Path path = Paths.get(this.getBaseLocation(), clientCode, resourcePath);
+
+					return this.createOrGetPath(path, urlResourcePath, fp, fileName, ovr, true);
+				},
+
+				(hasPermission, relativePath) -> {
+					// taking as an array so can update the path in future because Path data types behave as final
+					Path[] targetPath = new Path[1];
+					targetPath[0] = relativePath;
+
+					// updating the target path according to override condition
+					updatePathAccToOverrideCondition(fp, targetPath, filePath, ovr, fileName);
+
+					// just checking the fileTpe if it is an image or not. if not work here then put in fp==null && ovr
+					String fileType = null;
+					try {
+						fileType = Files.probeContentType(targetPath[0]);
+					} catch (IOException e2) {
+						return Mono.empty();
+					}
+
+					if (fileType == null || !fileType.startsWith("image/"))
+						return Mono.empty();
+					
+					return processImage(fp, targetPath, fileName, ovr, imageDetails, urlResourcePath, clientCode);
+
+				})
+				.switchIfEmpty(msgService.throwMessage(msg -> new GenericException(HttpStatus.BAD_REQUEST, msg),
+						FilesMessageResourceService.IMAGE_FILE_REQUIRED))
+				.contextWrite(Context.of(LogUtil.METHOD_NAME, "AbstractFilesResourceService.create"));
+	}
+	
+	private Mono<Void> updatePathAccToOverrideCondition(FilePart fp, Path[] targetPath, String filePath, boolean ovr,
+			String fileName) {
+		if (targetPath[0] != null && fp == null) {
+			targetPath[0] = Paths.get(this.getBaseLocation(), this.resolvePathWithClientCode(filePath).getT1());
+			if (!ovr) {
+				for (int i = 1; i <= 100; i++) {
```

Comment:
why the loop was being run for 100 size. Is there any reason?

Reply:
Ya like if there is already an image with name abc1 and User uploading a new image with same name so it won't override it but it will keep creating a new name by adding 1-100 number to the name and if the new name is not present it will add the file with that name.